authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,237 @@ func getActionName(verb string) string {
+			}
+
+			authInfoList, err = getAuthInfoListForWildcard(subRevReq, operationsMap)
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("Error which creating actions for checkaccess for Group: %s, Resource: %s, Verb: %s", subRevReq.ResourceAttributes.Group, subRevReq.ResourceAttributes.Resource, subRevReq.ResourceAttributes.Verb))
+			}
+
+		}
 	} else if subRevReq.NonResourceAttributes != nil {
-		authInfo.AuthorizationEntity.Id = path.Join(authInfo.AuthorizationEntity.Id, subRevReq.NonResourceAttributes.Path, getActionName(subRevReq.NonResourceAttributes.Verb))
+		authInfoSingle := azureutils.AuthorizationActionInfo{
+			IsDataAction: true,
+		}
+		authInfoSingle.AuthorizationEntity.Id = path.Join(clusterType, subRevReq.NonResourceAttributes.Path, getActionName(subRevReq.NonResourceAttributes.Verb))
+		authInfoList = append(authInfoList, authInfoSingle)
 	}
-	return authInfo
+	return authInfoList, nil
+}
+
+func getAuthInfoListForWildcard(subRevReq *authzv1.SubjectAccessReviewSpec, operationsMap azureutils.OperationsMap) ([]azureutils.AuthorizationActionInfo, error) {
+	var authInfoList []azureutils.AuthorizationActionInfo
+	var err error
+	if subRevReq.ResourceAttributes.Resource == "*" {
+		/*
+			This sections handles the following scenarios:
+
+			| Scenario              | Namespace is empty (Cluster scope call)                    | Namespace is not empty (NS scope)             |
+			------------------------ ------------------------------------------------------------  ----------------------------------------------
+			| Verb-*, Res-*, Group-*| All cluster and ns res with all verbs at clusterscope      | All ns resources at ns scope                  |
+
+			| Res-*, Group-*        | All cluster and ns res with specified verb at clusterscope | All ns res with specified verb at ns scope    |
+
+			| Verb-*, Res-*         | All cluster and ns res with all verbs under                | All ns res with all verbs under specified     |
+			|                       | specified apigroup at clusterscope                         | apigroup at nsscope                           |
+
+			| Resource - *          | All CS and NS Resources under specifed apigroup with       | All NS Resources under specifed apigroup with |
+			|                       | specified verb at cluster scope                            | specified verb at ns scope                    |
+		*/
+
+		filteredOperations := azureutils.NewOperationsMap()
+		if subRevReq.ResourceAttributes.Group == "*" {
+			// all resources under all apigroups
+			filteredOperations = operationsMap
+		} else if subRevReq.ResourceAttributes.Group != "" {
+			// all resources under specified apigroup
+			if value, found := operationsMap[subRevReq.ResourceAttributes.Group]; found {
+				filteredOperations[subRevReq.ResourceAttributes.Group] = value
```

Comment:
should we check that this is not empty instead of sending empty data actions in check access request?